perk-store-webapps/src/main/webapp/vue-app/components/perk-store/RefundDrawer.vue

```diff
@@ -334,25 +339,20 @@ export default {
           message: message,
         }}));
         if (!this.error){
-          this.showAlert(
-            'success', 
-            this.$t('exoplatform.wallet.metamask.message.transactionSent'),
-            null
-          );
           this.close();
         }
       } else {
         if (window.ethereum?.isMetaMask) {
           return searchUserOrSpaceObject(
-            this.product.receiverMarchand.id,
-            this.product.receiverMarchand.type
+            this.order.sender.id,
+            this.order.sender.type,
           )
             .then(wallet => {
               this.$emit('opened-transaction', true);
               const transactionParameters = {
                 to: this.contractDetails.address, 
                 from: this.walletAddress, 
-                data: this.contractDetails.contract.methods.transfer(this.order.sender.id, this.DecimalsAmount).encodeABI(),
+                data: this.contractDetails.contract.methods.transfer(wallet.address, this.DecimalsAmount).encodeABI(),
```

Comment:
Shouldn't this be the sender address instead of current user address ?

```suggestion
                data: this.contractDetails.contract.methods.transfer(this.order.sender.address, this.DecimalsAmount).encodeABI(),
```